Save spatial projections in netcdf files

hmkhatri opened the pull request:
The updated code saves spatial projections in netcdf files. This is in addition to data saved in .npy format. These code updates do not apply to some components (gia, greensmb, greendyn), which are only saved in .npy. Another pull request will address these components.

hmkhatri committed:
write lalon spatial projections in netcdf files

gregrmunday reviewed (changes requested):
This looks great, just a few suggestions:

1. Suggest removing line 230 as we don't need both `.npy` and `.nc`outputs
2. Compress the output file using something like
```python
encoding = {component: {"zlib": True, "complevel": 5}}
ds.to_netcdf(filename, encoding=encoding)
```
3. add a `source` and `description` attribute to the saved output, something like `xr_dataArray.attrs["source"] = "ProFSea-Climate v0.1"`, and `xr_dataArray.attrs["description"] = f"Regional {component} sea-level projections"`
4. Remove space around the `=` on line 238
5. Add space between `, 360` on line 236

hmkhatri committed:
Remove saving to .npy, add encoding to netcdf, save gia component in netcdf file

hmkhatri commented:
@gregrmunday Thanks for the review.

I have implemented suggested changes - encoding in netcdfs and removed saving to .npy files. I have also added code for saving GIA component in netcdf file, which was not there in the original pull request.

gregrmunday reviewed (approved):
Looks great to me, thanks Hemant!
